seclea_ai/internal/file_processor.py

```diff
@@ -22,15 +24,82 @@ class FileProcessor:
+        self._sender_q = Queue()
+
+    def writer(self, _storage_q: Queue):
+        self._storage_q = _storage_q
+        while not self._storage_q.empty():
+            obj = self._storage_q.get()
+
+            if obj:
+                if obj["function"] == "upload_dataset":
+                    self._save_dataset(
+                        obj["project"],
+                        obj["dataset"],
+                        obj["dataset_name"],
+                        obj["metadata"],
+                        obj["parent_hash"],
+                        obj["transformation"],
+                    )
+                if obj["function"] == "upload_training_run":
+                    self.upload_training_run(
+                        model=obj["model"],
+                        train_dataset=obj["train_dataset"],
+                        test_dataset=obj["test_dataset"],
+                        val_dataset=obj["val_dataset"],
+                        project=obj["project"],
+                    )
+                if obj["function"] == "_save_model_state":
+                    self._save_model_state(
+                        model=obj["model"],
+                        training_run_pk=obj["training_run_pk"],
+                        sequence_num=obj["sequence_num"],
+                        final=obj["final"],
+                        model_manager=obj["model_manager"],
+                    )
+
+    def sender(self):
+        while not self._sender_q.empty():
+            obj = self._sender_q.get()
+            if obj:
+                if obj["function"] == "_send_model_state":
+                    self._send_model_state(
+                        save_path=obj["save_path"],
+                        training_run_pk=obj["training_run_pk"],
+                        sequence_num=obj["sequence_num"],
+                        final=obj["final"],
+                    )
+                if obj["function"] == "_send_dataset":
+                    self._send_dataset(
+                        project=obj["project"],
+                        metadata=obj["metadata"],
+                        parent_hash=obj["parent_hash"],
+                        transformation=obj["transformation"],
+                        dataset_name=obj["dataset_name"],
+                        dataset_hash=obj["dataset_hash"],
+                        comp_path=obj["comp_path"],
+                        dataset_path=obj["dataset_path"],
+                    )
+                if obj["function"] == "_upload_training_run":
+                    self._upload_training_run(
+                        project=obj["project"],
+                        model=obj["model"],
+                        framework=obj["framework"],
+                        training_run_name=obj["training_run_name"],
+                        model_pk=obj["model_pk"],
+                        dataset_pks=obj["dataset_pks"],
+                        params=obj["params"],
+                    )
```

Comment:
This could all be written much more concisely by using dicts and unpacking to fill function arguments.

Reply:
Haresh did resolve this in his last commit.